components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pap/store/PAPPolicyStoreReader.java

```diff
@@ -74,14 +73,15 @@ public synchronized AbstractPolicy readPolicy(String policyId, PolicyFinder find
      */
     public PolicyDTO[] readAllLightPolicyDTOs() throws EntitlementException {
 
-        String[] resources;
-        resources = store.listPolicyIds().toArray(new String[0]);
+        List<String> policyIds = store.listPolicyIds();
 
-        List<PolicyDTO> policyDTOList = new ArrayList<>();
-
-        for (String resource : resources) {
-            PolicyDTO policyDTO = readLightPolicyDTO(resource);
-            policyDTOList.add(policyDTO);
+        List<PolicyDTO> policyDTOList = store.getPAPPolicies(policyIds);
+        for (PolicyDTO dto : policyDTOList) {
+            if (dto != null) {
+                dto.setPolicy(null);
+                dto.setAttributeDTOs(new AttributeDTO[0]);
+                dto.setPolicyEditorData(new String[0]);
```

Comment:
This logic also duplicated at the `readLightPolicyDTO()` we well, right? Shall we extract both into a private method?